oml/retrieval/retrieval_results.py

```diff
@@ -24,34 +26,46 @@
 class RetrievalResults:
     def __init__(
         self,
-        distances: FloatTensor,
-        retrieved_ids: LongTensor,
-        gt_ids: List[LongTensor] = None,
+        distances: Sequence[FloatTensor],
+        retrieved_ids: Sequence[LongTensor],
+        gt_ids: Sequence[LongTensor] = None,
     ):
         """
         Args:
-            distances: Sorted distances to the first ``top_n`` gallery items with the shape of ``[n_query, top_n]``.
-            retrieved_ids: Top N gallery ids retrieved for every query with the shape of ``[n_query, top_n]``.
-                Every element is within the range ``(0, n_gallery - 1)``.
-            gt_ids: Gallery ids relevant to every query, list of ``n_query`` elements where every element may
-                have an arbitrary length. Every element is within the range ``(0, n_gallery - 1)``
+            distances: Sorted distances from queries to the first gallery items with the size of ``n_query``.
+            retrieved_ids: First gallery indices retrieved for every query with the size of ``n_query``.
+                Every index is within the range ``(0, n_gallery - 1)``.
+            gt_ids: Gallery indices relevant to every query with the size of ``n_query``.
+                Every element is within the range ``(0, n_gallery - 1)``
 
         """
-        assert distances.shape == retrieved_ids.shape
-        assert distances.ndim == 2
+        for d, r in zip(distances, retrieved_ids):
+            if not (d[:-1] <= d[1:]).all():
```

Comment:
Logically it's okay to have empty distances and retrieved_ids (e.g. after filtering by threshold) for some queries, and it's funny that all these checks pass 😁 I was not sure about how 1 check should work for empty tensors.